ml/functions.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+
+import torch
+from torch import Tensor
+
+from ml.typing import TensorWithMask
+
+
+def concat(tensors: List[TensorWithMask], dim):
+    embeddings = []
+    masks = []
+
+    for tensor_with_mask in tensors:
+        embeddings.append(tensor_with_mask.tensor)
+        masks.append(tensor_with_mask.mask)
+
+    return TensorWithMask(
+        torch.cat(embeddings, dim=dim),
+        torch.cat(masks, dim=dim)
+    )
+
+
+def softmax(inputs: TensorWithMask, dim) -> Tensor:
```

Comment:
Let's put type for `dim` as well. 
```suggestion
def softmax(inputs: TensorWithMask, dim: int) -> Tensor:
```